main.rb

```diff
@@ -1,102 +1,146 @@
-#The Enumarable project: Creating own methods
+# The Enumarable project: Creating own methods
+
+# frozen_string_literal: true
+
+# rubocop:disable Metrics/ModuleLength
 module Enumerable
+  def my_each
+    return to_enum unless block_given?
 
-    def my_each
-      raise LocalJumpError, 'You have not given a block to the method' unless block_given?
-      for el in self
-        yield el
-      end
+    tmp = is_a?(Range) ? to_a : self
+    i = 0
+    result = []
+    while i < tmp.size
+      result << yield(tmp[i])
+      i += 1
     end
+    result
+  end
 
-    def my_each_with_index
-        raise LocalJumpError, 'You have not given a block to the method' unless block_given?
-        i = 0
-        for el in self
-            yield i,el
-            i+=1 
-        end
-    end
+  def my_each_with_index
+    return to_enum :my_each_with_index unless block_given?
 
-    def my_select
-      raise LocalJumpError, 'You have not given a block to the method' unless block_given?
-      selected_items = []
-      my_each { |i| selected_items.push(i) if yield(i) }
-      selected_items
+    result = []
+    k = 0
+    my_each do |v|
+      result << yield(k, v)
+      k += 1
     end
+    result
+  end
+
+  def my_select
+    return to_enum :my_select unless block_given?
 
-    def my_all?
-      raise LocalJumpError, 'You have not given a block to the method' unless block_given?
-      result = true
-      self.my_each do |e|
-        unless yield(e)
-          result = false
-          break
-        end
+    selected_items = []
+    my_each { |i| selected_items.push(i) if yield(i) }
+    selected_items
+  end
+
+  # rubocop:disable  Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+
+  def my_all?
+    return true if self == []
+
+    truth = true
+    if block_given?
+      (0...length).each do |i|
+        truth = yield(self[i])
+        return truth if truth == false
+      end
+    else
+      (0...length).each do |i|
+        return false if self[i].nil? || self[i] == false
       end
-      result
+      return true
     end
+    truth
+  end
```

Comment:
Let's implement `my_all?` as we have with `my_any?` and `my_none?`. 

### Find more details about [`all?` here](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F)